test/classes/sequelize-service.test.ts

```diff
@@ -210,14 +209,28 @@ describe('SequelizeService', function () {
       expect(user).to.have.keys(['email', 'id']); // ID is auto selected
     });
 
+    it('should only resolve types for what was selected (+ auto selected id)', async () => {
+      await userService.create({ email: 'foo', password: 'bar' });
+      const [ user ] = await userService.find({}, { select: ['email'] });
+      expect(user).to.have.keys(['email', 'id']); // ID is auto selected
+      expect(user).to.not.have.keys(['first_name']);
+      // Note: user.first_name should not even compile!
+    });
+
+    it('should by default select everything', async () => {
+      await userService.create({ email: 'foo', password: 'bar' });
+      const [ user ] = await userService.find({});
+      expect(user).to.have.keys(['id', 'email', 'first_name', 'last_name', 'date_of_birth', 'lucky_number', 'password', 'password_last_updated_at', 'updated_at', 'created_at']);
```

Comment:
@SylvainEstevez (Same here) `user.isAdult` compiles. That's not intended, right?